server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryFactory.java

```diff
@@ -75,26 +92,28 @@ public Directory newDirectory(String repositoryName, String indexUUID, ShardId s
             // Derive the path for data directory of SEGMENTS
             BlobPath dataPath = pathStrategy.generatePath(dataPathInput);
             RemoteDirectory dataDirectory = new RemoteDirectory(
-                blobStoreRepository.blobStore().blobContainer(dataPath),
-                blobStoreRepository::maybeRateLimitRemoteUploadTransfers,
-                blobStoreRepository::maybeRateLimitRemoteDownloadTransfers
+                blobStoreDataRepository.blobStore().blobContainer(dataPath),
+                blobStoreDataRepository::maybeRateLimitRemoteUploadTransfers,
+                blobStoreDataRepository::maybeRateLimitRemoteDownloadTransfers
             );
 
+            BlobStoreRepository blobStoreMetadataRepository = ((BlobStoreRepository) metadataRepository);
+            BlobPath metadataRepositoryBasePath = blobStoreMetadataRepository.basePath();
             RemoteStorePathStrategy.PathInput mdPathInput = RemoteStorePathStrategy.PathInput.builder()
-                .basePath(repositoryBasePath)
+                .basePath(metadataRepositoryBasePath)
                 .indexUUID(indexUUID)
                 .shardId(shardIdStr)
                 .dataCategory(SEGMENTS)
                 .dataType(METADATA)
                 .build();
             // Derive the path for metadata directory of SEGMENTS
             BlobPath mdPath = pathStrategy.generatePath(mdPathInput);
-            RemoteDirectory metadataDirectory = new RemoteDirectory(blobStoreRepository.blobStore().blobContainer(mdPath));
+            RemoteDirectory metadataDirectory = new RemoteDirectory(blobStoreMetadataRepository.blobStore().blobContainer(mdPath));
 
             // The path for lock is derived within the RemoteStoreLockManagerFactory
             RemoteStoreLockManager mdLockManager = RemoteStoreLockManagerFactory.newLockManager(
                 repositoriesService.get(),
-                repositoryName,
+                dataRepositoryName,
```

Comment:
should lock files be mapped to metadata repository as their access pattern seems similar to metadata files

Reply:
Right, the locks are actually held on metadata files so it makes sense to use metadataDirectory here. Let me think more on this.

Reply:
Made the change to use metadata repository for locks.